src/components/EmailItem/index.jsx

```diff
@@ -1,19 +1,51 @@
-const EmailItem = ({ name, date, subject, body, read, id }) => {
-  let readStatus = "border-gray-300 bg-gray-100 text-black"
-  if (read === "0") {
-    readStatus = " border-white bg-gray-600 text-white"
+import { useState } from "react"
+
+const EmailItem = ({
+  name,
+  date,
+  subject,
+  body,
+  read,
+  id,
+  setCurrentEmail,
+  setShowCloseButton,
+  setSelectedEmail,
+  selectedEmail,
+}) => {
+  let emailColor = "border-gray-300 bg-gray-100 text-black"
+  if (selectedEmail === id) {
+    emailColor = "bg-blue-500 text-white"
+  } else if (read === "0") {
+    emailColor = " border-white bg-gray-600 text-white"
+  }
+
+  const displayEmail = (e) => {
+    fetch(`https://email-client-api.dev.io-academy.uk/emails/${id}`)
+      .then((response) => response.json())
+      .then((data) => {
+        setCurrentEmail(data.data.email)
+        setShowCloseButton(true)
+        setEmailColor(id)
+      })
+  }
```

Comment:
You don't need to do this fetch here, you are passing in the `name`, `date`, `subject` and `body` fom the parent.

So you don't actually *do anything* with this data in this component.

You are passing down some state to hold the result of this function, passing that state up, and then passing that data down again into the `InboxDetails` component.

Consider which component actually needs the *full content* of an email. Where would you actually need to do this fetch?